api/types/access_request.go

```diff
@@ -642,6 +642,26 @@ func (u *AccessRequestUpdate) Check() error {
 	return nil
 }
 
+// RequestReasonMode can be either "required" or "optional". Empty-string is treated as "optional".
+// If a role has the request reason mode set to "required", then reason is required for all Access
+// Requests requesting roles or resources allowed by this role. It applies only to users who have
+// this role assigned.
+type RequestReasonMode string
+
+const (
+	RequestReasonModeRequired RequestReasonMode = "required"
+	RequestReasonModeOptional RequestReasonMode = "optional"
+)
+
+func (m RequestReasonMode) Required() bool {
```

Comment:
can we also add a validation for the accepted values? 
That would be great if people couldn't set incorrect values

Reply:
https://github.com/gravitational/teleport/pull/49256#issuecomment-2492145718